erpc/networks_test.go

```diff
@@ -1129,3 +1147,394 @@ func TestNetwork_WeightedRandomSelect(t *testing.T) {
+	pup1, err := upr.NewUpstream("prjA", &common.UpstreamConfig{
+		Type:     common.UpstreamTypeEvm,
+		Id:       "rpc1",
+		Endpoint: "http://rpc1.localhost",
+		Evm: &common.EvmUpstreamConfig{
+			ChainId: 123,
+		},
+	}, &log.Logger)
+	if err != nil {
+		t.Error(err)
+	}
+	cl1, err := clr.GetOrCreateClient(pup1)
+	if err != nil {
+		t.Error(err)
+	}
+	pup1.Client = cl1
+
+	pup2, err := upr.NewUpstream("prjA", &common.UpstreamConfig{
+		Type:     common.UpstreamTypeEvm,
+		Id:       "rpc2",
+		Endpoint: "http://rpc2.localhost",
+		Evm: &common.EvmUpstreamConfig{
+			ChainId: 123,
+		},
+	}, &log.Logger)
+	if err != nil {
+		t.Error(err)
+	}
+	cl2, err := clr.GetOrCreateClient(pup2)
+	if err != nil {
+		t.Error(err)
+	}
+	pup2.Client = cl2
+
+	ntw, err := NewNetwork(&log.Logger, "prjA", &common.NetworkConfig{
+		Failsafe: fsCfg,
+	}, rlr)
+	if err != nil {
+		t.Error(err)
+	}
+	ntw.Upstreams = []*upstream.Upstream{pup1, pup2}
+
+	fakeReq := upstream.NewNormalizedRequest(requestBytes)
+	resp, err := ntw.Forward(ctx, fakeReq)
+
+	if len(gock.Pending()) > 0 {
+		t.Errorf("Expected all mocks to be consumed, got %v left", len(gock.Pending()))
+		for _, pending := range gock.Pending() {
+			t.Errorf("Pending mock: %v", pending)
+		}
+	}
+
+	if err != nil {
+		t.Errorf("Expected nil error, got %v", err)
+		return
+	}
+
+	jrr, err := resp.JsonRpcResponse()
+	if err != nil {
+		t.Errorf("Expected nil error, got %v", err)
+		return
+	}
+
+	if jrr.Result == nil {
+		t.Errorf("Expected result, got nil")
+		return
+	}
+
+	// Debug logging
+	t.Logf("jrr.Result type: %T", jrr.Result)
+	t.Logf("jrr.Result content: %+v", jrr.Result)
+
+	// Type assertion with check
+	result, ok := jrr.Result.(map[string]interface{})
+	if !ok {
+		t.Errorf("Expected Result to be map[string]interface{}, got %T", jrr.Result)
+		return
+	}
+
+	fromHost, ok := result["fromHost"].(string)
+	if !ok {
+		t.Errorf("Expected fromHost to be string, got %T", result["fromHost"])
+		return
+	}
+
+	if fromHost != "rpc2" {
+		t.Errorf("Expected fromHost to be %v, got %v", "rpc2", fromHost)
+	}
+}
+
+func TestNetwork_ForwardEthGetLogsEmptyArrayResponseSuccess(t *testing.T) {
+	defer gock.Clean()
+	defer gock.DisableNetworking()
+	defer gock.DisableNetworkingFilters()
+
+	gock.EnableNetworking()
+
+	// Register a networking filter
+	gock.NetworkingFilter(func(req *http.Request) bool {
+		shouldMakeRealCall := strings.Split(req.URL.Host, ":")[0] == "localhost"
+		return shouldMakeRealCall
+	})
+
+	var requestBytes = json.RawMessage(`{"jsonrpc": "2.0","method": "eth_getLogs","params":[{"address":"0x1234567890abcdef1234567890abcdef12345678"}],"id": 1}`)
+
+	gock.New("http://rpc1.localhost").
+		Post("").
+		Reply(200).
+		JSON(json.RawMessage(`{"result":[]}`)).
+		Delay(500 * time.Millisecond)
+
+	gock.New("http://rpc2.localhost").
+		Post("").
+		Reply(200).
+		JSON(json.RawMessage(`{"result":{"hash":"0x64d340d2470d2ed0ec979b72d79af9cd09fc4eb2b89ae98728d5fb07fd89baf9","fromHost":"rpc2"}}`)).
```

Comment:
```suggestion
		JSON(json.RawMessage(`{"result":[{"logIndex":5,"fromHost":"rpc2"}]}`)).
```
Perhaps better to mimic array to be close to getLogs response?